### ISSUE TITLE
Why when I put Pretrained = False, yolov8 still transfer and freeze weights

### ISSUE THREAD

**Issue by quocanhnguyenBF** [ultralytics/ultralytics#14523]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Terminal output

WARNING ⚠️ 'ultralytics.yolo.v8' is deprecated since '8.0.136' and will be removed in '8.1.0'. Please use 'ultralytics.models.yolo' instead.
WARNING ⚠️ 'ultralytics.yolo.utils' is deprecated since '8.0.136' and will be removed in '8.1.0'. Please use 'ultralytics.utils' instead.
Note this warning may be related to loading older models. You can update your model to current structure with:
    import torch
    ckpt = torch.load("model.pt")  # applies to both official and custom models
    torch.save(ckpt, "updated-model.pt")

wandb: Currently logged in as: quocanh_nguyen (spfg). Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.17.4
wandb: Run data is saved locally in /mnt/c/Users/ai2/Documents/maritime_mcgriddles_v1/wandb/run-20240718_155904-mfnv23va
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run ...
wandb: ⭐️ View project at ...
wandb: 🚀 View run at ...
New https://pypi.org/project/ultralytics/8.2.59 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.0.178 🚀 Python-3.9.19 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3070, 8192MiB)
engine/trainer: task=detect, mode=train, model=yolov8n.pt, data=configs/data.yaml, epochs=3, patience=150, batch=-1, imgsz=1280, save=True, save_period=-1, cache=False, device=0, workers=8, project=None, name=None, exist_ok=True, pretrained=False, optimizer=Adam, verbose=True, seed=29, deterministic=True, single_cls=False, rect=True, cos_lr=False, close_mosaic=0, resume=False, amp=False, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=True, save_hybrid=False, conf=0.001, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, stream_buffer=False, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.3, hsv_v=0.3, degrees=3.0, translate=0.1, scale=0.1, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.4, mosaic=0.0, mixup=0.0, copy_paste=0.0, cfg=configs/config.yaml, tracker=botsort.yaml, save_dir=runs/detect/train
Overriding model.yaml nc=80 with nc=5

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]
 22        [15, 18, 21]  1    752287  ultralytics.nn.modules.head.Detect           [5, [64, 128, 256]]
Model summary: 225 layers, 3011823 parameters, 3011807 gradients

Transferred 319/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'

**Comment by Y-T-G** [comment]
To avoid transfer learning, use `.yaml` instead of `.pt`

**Comment by quocanhnguyenBF** [comment]
I did use .yaml as config. Im not sure what u mean

# Attach training configurations to experiment

yolov8_model_config: Final[str] = "configs/config.yaml"
yolov8_data_config: Final[str] = "configs/data.yaml"

with open(yolov8_model_config) as f:
    model_configs: dict = yaml.safe_load(f)
with open(yolov8_data_config) as f:
    data_configs: dict = yaml.safe_load(f)

wandb.init(project=proj_name, name=exp_name, job_type="training",  config=model_configs, tags=["model_name", "num_class"])
config = wandb.config

model = YOLO(model_configs.get("model", "yolov8n.pt"))
add_wandb_callback(model, enable_model_checkpointing=True)

# Log the config and data files as artifactss
log_artifact(yolov8_data_config, 'dataset')
log_artifact(yolov8_model_config, 'config')
log_artifact('./train.py', 'train.py')

model.train(cfg=yolov8_model_config, data=yolov8_data_config)  # start training
metrics = model.val()

**Comment by Y-T-G** [comment]
> model = YOLO(model_configs.get("model", "yolov8n.pt"))

You are using `.pt` here. Use `.yaml`.

**Comment by quocanhnguyenBF** [comment]
That model_configs is a dict pull from a yaml file. the string "yolov8n.pt" is the default string input if model_configs.get("model") is null. model_configs.get(model) is a strin path to my trained weight paths.
So maybe im misunderstanding something?

**Comment by pderrenger** [comment]
@quocanhnguyenBF thank you for the clarification. If `model_configs.get("model")` points to a `.pt` file, it will load pretrained weights by default. To ensure no weights are transferred, use a `.yaml` configuration file for the model architecture instead. This will build the model from scratch without loading any pretrained weights. If the issue persists, please verify it with the latest version of the Ultralytics package.

**Comment by Y-T-G** [comment]
If you use `yolov8n.pt`, it will use the pretrained model. If you use `yolov8n.yaml`, it will create a model from scratch.
